File: app/src/main/java/com/carpinelli/testeandroid/ui/invest/InvestPresenter.java
package com.carpinelli.testeandroid.ui.invest;

import android.util.Log;
import android.view.View;

import com.carpinelli.testeandroid.service.InvestService;
import com.carpinelli.testeandroid.service.RetrofitInitializer;
import com.carpinelli.testeandroid.service.dto.InvestmentSync;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class InvestPresenter implements MvpInvest.Presenter {

    private static final String TAG = InvestService.class.getSimpleName();

    private MvpInvest.View mvpView;

    public InvestPresenter(MvpInvest.View mvpView) {

        this.mvpView = mvpView;
    }

    @Override
    public void onStart() {

        if (mvpView != null) {

            Call<InvestmentSync> call = new RetrofitInitializer().getInvestService().getInvestment();

            call.enqueue(new Callback<InvestmentSync>() {
                @Override
                public void onResponse(Call<InvestmentSync> call, Response<InvestmentSync> response) {

                    mvpView.onInvestmentReady(response.body().getInvestment());
                }

                @Override
                public void onFailure(Call<InvestmentSync> call, Throwable t) {
                    Log.d(TAG, "onFailure: ");
                }
            });

        }
    }

    @Override
    public void onAttach(Object mvpView) {

    }

}

File: source/app/src/main/java/br/com/andreyneto/testesantander/ui/invest/InvestPresenter.kt
package br.com.andreyneto.testesantander.ui.invest

import android.content.Context
import android.widget.Toast
import br.com.andreyneto.testesantander.model.InvestResponse
import br.com.andreyneto.testesantander.service.ApiService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class InvestPresenter(
        val view: InvestContract.View
): InvestContract.Presenter {

    override fun toast(ctx: Context, text: String) {
        Toast.makeText(ctx, text, Toast.LENGTH_SHORT).show()
    }

    override fun start() {
        getData()
    }

    private val apiService = ApiService()

    init {
        view.setPresenter(this)
    }

    override fun getData() {
        apiService.getApi().fund.enqueue(object : Callback<InvestResponse> {
            override fun onResponse(call: Call<InvestResponse>, response: Response<InvestResponse>) {
                if (response.isSuccessful) {
                    view.showData(response.body()?.screen!!)
                }
            }

            override fun onFailure(call: Call<InvestResponse>, t: Throwable) {

            }
        })
    }
}